app/lib/dartdoc/backend.dart

```diff
@@ -133,10 +133,33 @@ class DartdocBackend {
     return versions.map((pv) => pv.version).take(limit).toList();
   }
 
+  Future _uploadWithRetry(
+      String objectName, Stream<List<int>> openStream()) async {
+    await retryAsync(
+      () async {
+        final sink =
+            _storage.write(objectName, contentType: contentType(objectName));
```

Comment:
please consider supplying the content-length as well

Reply:
Will do in a follow-up (but first it will get a bit of refactoring for another PR).